packages/perseus/src/widgets/radio/radio-component.new.tsx

```diff
@@ -164,78 +134,50 @@ class Radio extends React.Component<Props> implements Widget {
-                    ...this.props.linterContext,
+                    ...linterContext,
                     // @ts-expect-error - TS2322 - Type '{ blockHighlight: true; contentType: string; highlightLint: boolean; paths: readonly string[]; stack: readonly string[]; }' is not assignable to type 'LinterContextProps'.
                     blockHighlight: true,
                 }}
-                strings={this.context.strings}
+                strings={strings}
             />
         );
     };
 
-    // TODO(LP-10672): I think this might be unused right now. I can't find anywhere
-    // that we pass a value to `.focus()` and it seems to have been used for
-    // adding hints when editing.
-    // See: https://github.com/Khan/perseus/blame/e18582b4b69959270b90e237ef1813899711ddfa/src/widgets/radio.js#L169
-    focus(choiceIndex?: number | null): boolean {
-        // eslint-disable-next-line @typescript-eslint/strict-boolean-expressions
-        if (this.focusFunction) {
-            return this.focusFunction(choiceIndex);
-        }
-
-        return false;
-    }
-
-    // lets BaseRadio register a focus callback so widget
-    // can focus an individual choice
-    registerFocusFunction(fun: FocusFunction): void {
-        this.focusFunction = fun;
-    }
-
-    // When `BaseRadio`'s `onChange` handler is called, indicating a change in
-    // our choices' state, we need to call our `onChange` handler in order to
-    // persist those changes in the item's Perseus state.
-    //
-    // So, given the new values for each choice, construct the new
-    // `choiceStates` objects, and pass them to `this.props.onChange`.
-    //
-    // `newValueLists` is an object with two keys: `checked` and `crossedOut`.
-    // Each contains an array of boolean values, specifying the new checked and
-    // crossed-out value of each choice.
-    //
-    // NOTE(mdr): This method expects to be auto-bound. If this component is
-    //     converted to an ES6 class, take care to auto-bind this method!
-    updateChoices: (
+    /* Update the choice states for the widget upon user interaction.
+    This function is called by the BaseRadio component when the user selects
+    or deselects a choice.
+    */
+    const updateChoices = (
         newValueLists: Readonly<{
             checked: ReadonlyArray<boolean>;
             crossedOut: ReadonlyArray<boolean>;
         }>,
-    ) => void = (newValueLists) => {
-        const {choiceStates, choices} = this.props;
+    ) => {
+        const {choiceStates} = props;
 
         // Construct the baseline `choiceStates` objects. If this is the user's
         // first interaction with the widget, we'll need to initialize them to
         // new objects with all fields set to the default values. Otherwise, we
         // should clone the old `choiceStates` objects, in preparation to
         // mutate them.
+        // eslint-disable-next-line functional/immutable-data
         const newChoiceStates = choiceStates
```

Comment:
This logic wasn't changed, but the suppressions are required for our new linter.